luigi/worker.py

```diff
@@ -238,8 +238,11 @@ class DequeQueue(collections.deque):
     deque wrapper implementing the Queue interface.
     """
 
-    put = collections.deque.append
-    get = collections.deque.pop
+    def put(self, *args, **kwargs):
```

Comment:
Great advice, thank Dave.

The PR has been updated.

class DequeQueue(collections.deque):
    """
    deque wrapper implementing the Queue interface.
    """

    def put(self, obj, block=None, timeout=None):
        del block, timeout
        return self.append(obj)

    def get(self, block=None, timeout=None):
        del block, timeout
        return self.pop()


On Tue, Nov 10, 2015 at 10:47 AM, Dave Buchfuhrer <notifications@github.com>
wrote:

> In luigi/worker.py
> <https://github.com/spotify/luigi/pull/1382#discussion_r44421800>:
>
> > @@ -238,8 +238,11 @@ class DequeQueue(collections.deque):
> >      deque wrapper implementing the Queue interface.
> >      """
> >
> > -    put = collections.deque.append
> > -    get = collections.deque.pop
> > +    def put(self, *args, **kwargs):
>
> It only needs to handle timeouts if it's using multiple processes. The
> worker will periodically check for new work while waiting for jobs to
> finish. If you're running with only one process, the timeout doesn't mean
> anything. Get and put will always return immediately. It's correct to throw
> out the timeout argument, but it would also be reasonable not to
> indiscriminately ignore any keyword argument and just match the signatures
> of the multiprocessing.Queue functions:
>
> put(obj[, block[, timeout]])
> get([block[, timeout]])
>
> Just adding optional block and timeout arguments that you ignore looks
> more intentional and makes it more clear that you're not throwing out
> important arguments. If someone were to call DequeQueue().put(obj=x), your
> code would throw out the obj argument and result in an error.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/spotify/luigi/pull/1382/files#r44421800>.
>